godot-macros/src/lib.rs

```diff
@@ -823,6 +824,58 @@ pub fn derive_godot_class(input: TokenStream) -> TokenStream {
 /// [`TransferMode`]: ../classes/multiplayer_peer/struct.TransferMode.html
 /// [`RpcConfig`]: ../register/struct.RpcConfig.html
 ///
+/// # Lifecycle functions with custom receivers
+///
+/// Lifecycle functions, similarly to user-defined `#[func]`s, can be annotated with `#[func(gd_self)]` to use `Gd<Self>` receiver and
+/// avoid binding the instance.
```

Comment:
"Lifecycle functions" might not be a well-known term on its own... I would add something like "inside `I*` interface impls". The "similarly to user-defined..." part could maybe be its own sentence if it gets too long.